server/src/main/java/org/apache/druid/server/QueryResource.java

```diff
@@ -83,6 +83,7 @@
 
   protected static final int RESPONSE_CTX_HEADER_LEN_LIMIT = 7 * 1024;
```

Comment:
Why didn't you move this to `ResponseContext`?

Reply:
sounds good

Reply:
However, this is not a `ResponseContext` feature, it is more like a `QueryResource` limitation (also as `HEADER_RESPONSE_CONTEXT` http-header name). So decided to leave it as is, let me know if that explanation doesn't correlate with your vision.

Reply:
Please add a Javadoc comment explaining this.

Reply:
Added comments for length limit and http header name